src/core/service.c

```diff
@@ -3550,13 +3556,37 @@ static int service_dispatch_exec_io(sd_event_source *source, int fd, uint32_t ev
 
                         return 0;
                 }
+                if (n == 1) {
+                        /* Backward compatibility block: a single byte was read, which means somehow an old
+                         * executor before this logic was introduced sent the notification, so there is no
+                         * timestamp (reset it). */
```

Comment:
how could that possibly happen? you pin the executor binary...

Reply:
Paranoia. It would completely break Type=exec services if it was wrong, and it's very cheap to be on the safe side. EG: something being activated during a downgrade from 256 to 254 (no executor at all)